### PR TITLE
Adding Orderbook fill functionality

### PR DESCRIPTION
**Pull request by Tanz0rz**

### CODE REVIEW

File: orderbook/models/orderbook_types_manual.go [hunk 1]
```diff
@@ -1,34 +1,113 @@
 package models
 
-import "time"
+import (
+	"math/big"
+	"time"
+)
```
Comment (EnoRage) [inline orderbook/models/orderbook_types_manual.go:6]: Lets keep structs and methods in the same package, let's put all the models into orderbook pkg

File: internal/orderbook/limitorder.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func CreateLimitOrderMessage(orderRequest models.CreateOrderParams, makerTraits
 		Extension:     "0x",
 	}
 
-	aggregationRouter, err := constants.Get1inchRouterFromChainId(orderRequest.ChainId)
+	aggregationRouter, err := constants.Get1inchRouterFromChainId(chainId)
```
Comment (EnoRage) [inline internal/orderbook/limitorder.go:80]: Let's move it to web3data.go insidue orderbook, no any sense to have limit order specific funcs here

**Review by EnoRage — CHANGES_REQUESTED** [review]
Just 2 mentioned moments and LGTM

File: internal/orderbook/limitorder.go (hunk 2 above)
Comment (Tanz0rz) [reply internal/orderbook/limitorder.go:80]: Here is an updated workspace to show the advantages from a discoverability perspective

image

The `orderbook` directory now holds three top-level files and gives the user clear direction as to what methods will be relevant to them while creating orderbook calls. The `models` directory serves two purposes here: It collapses all models files into one line-item and additionally hides the `validate.go` file which is also not relevant to the user (but must be next to the model files with the current implementation of being functions attached to the structs).

This lets us move all helper functions away from the users' view entirely while they look at what methods are available to call. A similar functional equivalent would be to use private functions without the `internal` directory, but that would still clutter the files that hold the public methods (the ones the user will be testing/using in their project)

File: orderbook/models/orderbook_types_manual.go (hunk 1 above)
Comment (Tanz0rz) [reply orderbook/models/orderbook_types_manual.go:6]: The advantage of keeping these in a sub-folder is we can maintain a cleaner viewing space for people looking at our code when using the API. I expand on this further in the other comment.

File: orderbook/api.go [hunk 3]
```diff
@@ -44,9 +44,9 @@ func (api *api) CreateOrder(ctx context.Context, params models.CreateOrderParams
 		Nonce:              params.SeriesNonce.Int64(),
 		Series:             0, // TODO: Series 0 always?
 	}
-	makerTraits := orderbook.BuildMakerTraits(buildMakerTraitsParams)
+	makerTraits := internal.BuildMakerTraits(buildMakerTraitsParams)
 
-	order, err := orderbook.CreateLimitOrderMessage(params, makerTraits)
+	order, err := internal.CreateLimitOrderMessage(params, makerTraits, int(api.chainId))
```
Comment (EnoRage) [inline orderbook/api.go:49]: Let's split API and order message into separate functions and allow user to Build order message on it's own and to send it to API when it needed

File: orderbook/models/orderbook_types_manual.go (hunk 1 above)
Comment (EnoRage) [reply orderbook/models/orderbook_types_manual.go:6]: It does not make any sense to keep it in different packages, it breaks the common sense and does not make anything cleaner

**Review by EnoRage — CHANGES_REQUESTED** [review]

File: orderbook/internal/normalization.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	}, nil
+}
+
+func AddressStringToBigInt(addressString string) *big.Int {
+	address := common.HexToAddress(addressString)
+	addressBytes := new(big.Int).SetBytes(address.Bytes())
+	return addressBytes
+}
+
+func HexStringToBigInt(hexStr string) (*big.Int, error) {
```
Comment (EnoRage) [inline orderbook/internal/normalization.go:58]: maybe we can use hexutil.DecodeBig instead?

**Pull request merged**